### DISCUSSION TITLE
FastAPI incompatible with Airflow due to Flask-AppBuilder's email-validator constraint

### DISCUSSION THREAD

**Discussion by nickolasrm** [apache/airflow#37884] [Q&A]
### Apache Airflow version

2.8.2

### What happened?

it's not possible to use `fastapi` with `airflow` because airflow depends on `flask-appbuilder`. The `appbuilder` dependencies require `email-validator<2`, which is incompatible with `pydantic[email] `, thus making `fastapi` incompatible.

The new version of the `appbuilder` (^4.4) fixes this problem by relaxing `email-validator` constraint and doesn't seem to introduce any breaking changes to the airflow.

### How to reproduce

Try installing `pip install "fastapi[all]>=0.9" "apache-airflow==2.8.2"`

### Operating System

Ubuntu 22.04

### Deployment

Official Apache Airflow Helm Chart

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by raphaelauv** [comment]
don't you just need the airflow clients ?

https://pypi.org/project/apache-airflow-client/

**Reply by nickolasrm** [reply]
Not actually. Airflow and fastapi are both installed in the same environment because we use a monorepo with centralized dependencies. This is just our use case, but I think reducing the number of unnecessary constraints could be a good thing for airflow in general. By comparing FAB versions, it doesn't seem to introduce any bugs.

https://github.com/dpgaspar/Flask-AppBuilder/compare/v4.3.11...v4.4.1

**Reply by raphaelauv** [reply]
Monorepo doesn't mean single venv

**Reply by Taragolis** [reply]
@nickolasrm If you would like to help with testing new version FAB and it integration with Airflow you might bump version in FAB provider

https://github.com/apache/airflow/blob/5a0be392e66f8e5426ba3478621115e92fcf245b/airflow/providers/fab/provider.yaml#L42-L47

Please Note: This changes will reflected only in Airflow 2.9+ (when it released), in version below it is direct dependency for Airflow

**Reply by potiuk** [reply]
Note. That we are currently bound to a specific version of FAB (it's historical) - every time we bound to new version we have to incorporate changes from security manager that are coming with the new version. If you would like to help with bumping it - then what @Taragolis wrote - bumping version and copying all the changes from security manager are most welcome, and indeed, it will only be available in Airflow 2.9+. If you want to help similarly as other users - you are more than open to open a PR for that - like htis on https://github.com/apache/airflow/pull/32054 . I checked and there are some changes in security manager that should be ported. 

Would you like to help with that @nickolasrm ?
